docs/v3/how-to-guides/workflows/custom-metadata.mdx

```diff
@@ -0,0 +1,64 @@
+---
+title: How to customize a workflow's metadata
+sidebarTitle: Customize workflow metadata
+description: The simplest way to give your workflow some personality
+---
+
+Give your workflow a name with the `name` parameter:
+
+```python
+from prefect import flow
+
+
+@flow(name="My Flow")
+def my_workflow() -> str:
+    return "Hello, world!"
+```
+
+If you don't provide a name, Prefect uses the flow function name.
+
+Give your workflow a description with the `description` parameter:
+
+```python
+from prefect import flow
+
+
+@flow(name="My Flow", description="My flow with a name and description")
+def my_workflow() -> str:
+    return "Hello, world!"
+```
+
+If no description is provided, a flow function's docstring is used as the description.
+
+Give a custom name to each flow run for a flow with the `flow_run_name` parameter:
+
+```python
+import datetime
+from prefect import flow
+
+
+@flow(flow_run_name="hello-{name}-on-{date:%A}")
+def my_workflow(name: str, date: datetime.datetime):
+    return f"I said hello to {name} on {date:%A}!"
+```
+
+Inputs to the flow function are available as variables in the `flow_run_name` template string.
+
+You can also provide a `flow_run_name` as a function that returns a string:
+
+```python
+import datetime
+from prefect import flow
+
+
+def generate_flow_run_name():
+    date = datetime.datetime.now(datetime.timezone.utc)
+    return f"{date:%A}-is-a-nice-day"
+
+
+@flow(flow_run_name=generate_flow_run_name)
+def my_workflow(name: str):
+    return f"Hello, {name}!"
+```
+
```

Comment:
i think this needs a warning admonition that run names will be updated only after the run has begun (entered "Running"). The scheduler will not currently respect these names at run creation time.

Reply:
Added an info admonition at the bottom of the page.